primitives/trie/src/storage_proof.rs

```diff
@@ -114,6 +139,17 @@ impl<H: Hasher> From<StorageProof> for crate::MemoryDB<H> {
 	}
 }
 
+impl<H: Hasher> From<&StorageProof> for crate::MemoryDB<H> {
+	fn from(proof: &StorageProof) -> Self {
+		let mut db = crate::MemoryDB::default();
```

Comment:
Please make the implementation above use this one `(&self).into()` should do it.